### PR TITLE
POL-165 - Update AWS policies for credential Service 1

### PR DESCRIPTION
**Pull request by praut1**
### Description

Modified AWS policies to support credential service

### Issues Resolved

Modified AWS policies to support credential service

### Contribution Check List

- [ ] All tests pass.
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [ ] New functionality has been documented in CHANGELOG.MD

### Running Template links and screenshots

**AWS Disallowed Regions**
https://governance.rightscale.com/org/1105/projects/60073/applied-policies/5e1874b3513cd00001eaecae

image

**AWS Idle Compute Instances**
https://governance.rightscale.com/org/1105/projects/60073/applied-policies/5e186ed9828e5b00011722dd

image

**AWS Long Stopped Instances**
https://governance.rightscale.com/org/1105/projects/60073/applied-policies/5e18703a828e5b00011722e1

image

**AWS Publicly Accessible RDS Instances**
https://governance.rightscale.com/org/1105/projects/60073/applied-policies/5e186e51828e5b00011722dc

image

Out of 6 policies which were selected for end to end testing, pasted the links of applied policies and screenshots for 4 policies.
As discussed, AWS Internet facing ELBs policy syntax has been reviewed internally.
AWS object storage optimization policy changes have not been committed as testing the same has given 501 error for which ticket has been raised to resolve the issue(PHX-1971).

### CODE REVIEW

File: compliance/aws/disallowed_regions/README.md [hunk 1]
```diff
@@ -1,37 +1,38 @@
...
 - Terminate reported instances after approval.
 
-### Required RightScale Roles
- 
+## Required RightScale Roles
+
 - policy_designer
 - policy_manager
 - policy_publisher
 - credential_viewer
```
Comment (cdwilhelm) [inline compliance/aws/disallowed_regions/README.md]: please remove the Rightscale roles

File: cost/aws/reserved_instances/coverage/README.md [hunk 2]
```diff
@@ -1,33 +1,35 @@
-## Reserved Instances Coverage Policy Template
+# Reserved Instances Coverage Policy Template
```
Comment (cdwilhelm) [inline cost/aws/reserved_instances/coverage/README.md]: Please remove **policy template**

File: cost/aws/reserved_instances/recommendations/README.md [hunk 3]
```diff
@@ -1,46 +1,43 @@
-## AWS Reserved Instances Recommendation Policy Template
+# AWS Reserved Instances Recommendation Policy Template
```
Comment (cdwilhelm) [inline cost/aws/reserved_instances/recommendations/README.md]: Please remove **Policy Template**

File: cost/aws/s3_bucket_size/README.md [hunk 4]
```diff
@@ -1,33 +1,35 @@
-## AWS Bucket Size Check Policy Template
+# AWS Bucket Size Check Policy Template
```
Comment (cdwilhelm) [inline cost/aws/s3_bucket_size/README.md]: Please remove **Policy Template**

File: operational/dbaas/aws/rds_backup/README.md [hunk 5]
```diff
@@ -1,34 +1,35 @@
-## AWS RDS Backup Policy Template
+# AWS RDS Backup Policy Template
```
Comment (cdwilhelm) [inline operational/dbaas/aws/rds_backup/README.md]: Please remove Policy Template

File: security/aws/loadbalancer_internet_facing/README.md [hunk 6]
```diff
@@ -1,59 +1,68 @@
-## Check for internet-facing ELBs & ALBs 
- 
-### What it does
+# Check for internet-facing ELBs & ALBs
```
Comment (cdwilhelm) [inline security/aws/loadbalancer_internet_facing/README.md]: Change to **AWS Internet-facing ELBs & ALBs**

File: compliance/aws/disallowed_regions/README.md (hunk 1 above)
Comment (praut1) [reply compliance/aws/disallowed_regions/README.md]: Removed the Rightscale roles

File: cost/aws/reserved_instances/coverage/README.md (hunk 2 above)
Comment (praut1) [reply cost/aws/reserved_instances/coverage/README.md]: Removed policy template from header

File: cost/aws/reserved_instances/recommendations/README.md (hunk 3 above)
Comment (praut1) [reply cost/aws/reserved_instances/recommendations/README.md]: Removed policy template from header

File: cost/aws/s3_bucket_size/README.md (hunk 4 above)
Comment (praut1) [reply cost/aws/s3_bucket_size/README.md]: Removed policy template from header

File: operational/dbaas/aws/rds_backup/README.md (hunk 5 above)
Comment (praut1) [reply operational/dbaas/aws/rds_backup/README.md]: Removed policy template from header

File: security/aws/loadbalancer_internet_facing/README.md (hunk 6 above)
Comment (praut1) [reply security/aws/loadbalancer_internet_facing/README.md]: Changed the header name to AWS Internet-facing ELBs & ALBs

File: cost/aws/object_storage_optimization/README.md [hunk 7]
```diff
@@ -1,29 +1,24 @@
-## AWS Object Storage Optimization
- 
-### What it does
+# AWS Object Storage Optimization
```
Comment (cdwilhelm) [inline cost/aws/object_storage_optimization/README.md]: This file is missing the Prerequisite section and the AWS Permissions section should not include details about Cloud management.  See the other README files in the PR for good examples.

**Comment by cdwilhelm** [comment]
@praut1   also please see the Danger errors.

File: cost/aws/object_storage_optimization/README.md [hunk 8]
```diff
@@ -1,29 +1,24 @@
...
-### Required RightScale Roles
- 
-- Cloud Management - The `credential_viewer`,`observer` roles
-- Cloud Management - The `policy_designer`, `policy_manager` & `policy_publisher` roles
+- *Email addresses to notify* - Email addresses of the recipients you wish to notify when new incidents are created
+- *Days since last modified to move to Glacier* - Move to glacier after days last modified - leave blank to skip moving
+- *Days since last modified to move to Deep Archive* - Move to glacier deep archive after days last modified- leave blank to skip moving
+- *Exclude Tag* - List of tags that will exclude s3 objects from being evaluated by this policy. Multiple tags are evaluated as an 'OR' condition. Tag keys or key/value pairs can be listed. Example: 'test,env=dev'
 
-### Enable delete action
```
[the remark this answers is not in the document]
Comment (praut1) [reply cost/aws/object_storage_optimization/README.md]: Recovered Enable delete section

File: cost/aws/object_storage_optimization/README.md (hunk 7 above)
Comment (praut1) [reply cost/aws/object_storage_optimization/README.md]: Changes have been placed in all the 12 AWS policies.

File: security/aws/rds_publicly_accessible/README.md [hunk 9]
```diff
@@ -1,75 +1,81 @@
...
 
-### Enable delete action
+- Sends an email notification
+- Disable the publicly accessible RDS instances after approval
+- Delete publicly accessible RDS instances after approval (optional)
 
 Perform below steps to enable delete action.
 
-- Edit the file [AWS_Publicly_Accessible_RDS_Instances](https://github.com/rightscale/policy_templates/tree/master/security/aws/rds_publicly_accessible/AWS_Publicly_Accessible_RDS_Instances.pt)
+- Edit the file [AWS_Publicly_Accessible_RDS_Instances](https://github.com/flexera/policy_templates/tree/master/security/aws/rds_publicly_accessible/aws_publicly_accessible_rds_instances.pt)
```
Comment (cdwilhelm) [inline security/aws/rds_publicly_accessible/README.md]: to fix the 404 error in the danger file for this url remove this part of the link **https://github.com/flexera/policy_templates/tree/master**

File: compliance/aws/disallowed_regions/aws_disallowed_regions.pt [hunk 10]
```diff
@@ -43,11 +49,12 @@ end
...
 ###############################################################################
 
-auth "auth_aws", type: "aws" do
-  version "4"
-  service "ec2"
-  access_key cred("AWS_ACCESS_KEY_ID")
-  secret_key cred("AWS_SECRET_ACCESS_KEY")
+#authenticate with AWS
+credentials "auth_aws" do
+  schemes "aws"
```
Comment (cdwilhelm) [inline compliance/aws/disallowed_regions/aws_disallowed_regions.pt]: We need to also use the aws_sts scheme.  the schemes should be "aws","aws_sts"

File: compliance/aws/long_stopped_instances/aws_long_stopped_instances.pt [hunk 11]
```diff
@@ -42,18 +48,12 @@ end
...
-end
-
-auth "monitoring_auth_aws", type: "aws" do
-  version 4
-  service "monitoring"
-  access_key cred('AWS_ACCESS_KEY_ID')
-  secret_key cred('AWS_SECRET_ACCESS_KEY')
+#authenticate with AWS
+credentials "auth_aws" do
+  schemes "aws"
```
Comment (cdwilhelm) [inline compliance/aws/long_stopped_instances/aws_long_stopped_instances.pt]: We need to also use the aws_sts scheme.  the schemes should be "aws","aws_sts"

File: cost/aws/burstable_instance_cloudwatch_credit_utilization/aws_burstable_instance_cloudwatch_credit_utilization.pt [hunk 12]
```diff
@@ -59,18 +65,12 @@ end
...
-end
-
-auth "monitoring_auth_aws", type: "aws" do
-  version 4
-  service "monitoring"
-  access_key cred('AWS_ACCESS_KEY_ID')
-  secret_key cred('AWS_SECRET_ACCESS_KEY')
+#authenticate with AWS
+credentials "auth_aws" do
+  schemes "aws"
```
Comment (cdwilhelm) [inline cost/aws/burstable_instance_cloudwatch_credit_utilization/aws_burstable_instance_cloudwatch_credit_utilization.pt]: We need to also use the aws_sts scheme.  the schemes should be "aws","aws_sts"

File: cost/aws/elb/clb_unused/aws_delete_unused_clb.pt [hunk 13]
```diff
@@ -34,11 +40,12 @@ end
...
 ###############################################################################
 
-auth "auth_aws", type: "aws" do
-  version 4
-  service "elasticloadbalancing"
-  access_key cred('AWS_ACCESS_KEY_ID')
-  secret_key cred('AWS_SECRET_ACCESS_KEY')
+#authenticate with AWS
+credentials "auth_aws" do
+  schemes "aws"
```
Comment (cdwilhelm) [inline cost/aws/elb/clb_unused/aws_delete_unused_clb.pt]: We need to also use the aws_sts scheme.  the schemes should be "aws","aws_sts"

File: cost/aws/object_storage_optimization/aws_object_storage_optimization.pt [hunk 14]
```diff
@@ -47,11 +53,12 @@ end
...
 ###############################################################################
 
-auth "auth_aws", type: "aws" do
-  version "4"
-  service "s3"
-  access_key cred("AWS_ACCESS_KEY_ID")
-  secret_key cred("AWS_SECRET_ACCESS_KEY")
+#authenticate with AWS
+credentials "auth_aws" do
+  schemes "aws"
```
Comment (cdwilhelm) [inline cost/aws/object_storage_optimization/aws_object_storage_optimization.pt]: We need to also use the aws_sts scheme.  the schemes should be "aws","aws_sts"

File: cost/aws/reserved_instances/recommendations/aws_reserved_instance_recommendations.pt [hunk 15]
```diff
@@ -71,14 +84,22 @@ parameter "param_email" do
...
-  region "us-east-1"
-  access_key cred("AWS_ACCESS_KEY_ID")
-  secret_key cred("AWS_SECRET_ACCESS_KEY")
+###############################################################################
+# Authentication
+###############################################################################
+
+#authenticate with AWS
+credentials "auth_aws" do
+  schemes "aws"
```
Comment (cdwilhelm) [inline cost/aws/reserved_instances/recommendations/aws_reserved_instance_recommendations.pt]: We need to also use the aws_sts scheme.  the schemes should be "aws","aws_sts"

File: cost/aws/reserved_instances/coverage/reserved_instance_coverage.pt [hunk 16]
```diff
@@ -27,14 +37,22 @@ parameter "param_historical_number_of_days_in_past" do
...
-  region "us-east-1"
-  access_key cred("AWS_ACCESS_KEY_ID")
-  secret_key cred("AWS_SECRET_ACCESS_KEY")
+###############################################################################
+# Authentication
+###############################################################################
+
+#authenticate with AWS
+credentials "auth_aws" do
+  schemes "aws"
```
Comment (cdwilhelm) [inline cost/aws/reserved_instances/coverage/reserved_instance_coverage.pt]: We need to also use the aws_sts scheme.  the schemes should be "aws","aws_sts"

File: operational/dbaas/aws/rds_backup/aws_rds_backup.pt [hunk 17]
```diff
@@ -28,13 +42,21 @@ parameter "param_backup_window" do
...
-  service 'rds'
-  access_key cred('AWS_ACCESS_KEY_ID')
-  secret_key cred('AWS_SECRET_ACCESS_KEY')
+###############################################################################
+# Authentication
+###############################################################################
+
+#authenticate with AWS
+credentials "auth_aws" do
+  schemes "aws"
```
Comment (cdwilhelm) [inline operational/dbaas/aws/rds_backup/aws_rds_backup.pt]: We need to also use the aws_sts scheme.  the schemes should be "aws","aws_sts"

File: security/aws/loadbalancer_internet_facing/aws_internet-facing_elbs.pt [hunk 18]
```diff
@@ -35,11 +41,12 @@ end
...
 ###############################################################################
 
-auth "auth_aws", type: "aws" do
-  version 4
-  service "elasticloadbalancing"
-  access_key cred('AWS_ACCESS_KEY_ID')
-  secret_key cred('AWS_SECRET_ACCESS_KEY')
+#authenticate with AWS
+credentials "auth_aws" do
+  schemes "aws"
```
Comment (cdwilhelm) [inline security/aws/loadbalancer_internet_facing/aws_internet-facing_elbs.pt]: We need to also use the aws_sts scheme.  the schemes should be "aws","aws_sts"

File: security/aws/rds_publicly_accessible/aws_publicly_accessible_rds_instances.pt [hunk 19]
```diff
@@ -35,11 +41,12 @@ end
...
 ###############################################################################
 
-auth "auth_aws", type: "aws" do
-  version "4"
-  service "rds"
-  access_key cred("AWS_ACCESS_KEY_ID")
-  secret_key cred("AWS_SECRET_ACCESS_KEY")
+#authenticate with AWS
+credentials "auth_aws" do
+  schemes "aws"
```
Comment (cdwilhelm) [inline security/aws/rds_publicly_accessible/aws_publicly_accessible_rds_instances.pt]: We need to also use the aws_sts scheme.  the schemes should be "aws","aws_sts"

File: compliance/aws/disallowed_regions/aws_disallowed_regions.pt [hunk 20]
```diff
@@ -1,10 +1,16 @@
 name "AWS Disallowed Regions"
 rs_pt_ver 20180301
 type "policy"
-short_description "Report on any instances that are in a disallowed region with the option to Terminate them. \n See the [README](https://github.com/flexera/policy_templates/tree/master/aws/disallowed_regions) and [docs.rightscale.com/policies](http://docs.rightscale.com/policies/) to learn more."
-long_description "Version: 1.0"
+short_description "Check for instances that are in a disallowed region with the option to terminate them. \n See the [README](https://github.com/flexera/policy_templates/tree/master/aws/disallowed_regions) and [docs.rightscale.com/policies](https://docs.rightscale.com/policies/) to learn more."
```
Comment (cdwilhelm) [inline compliance/aws/disallowed_regions/aws_disallowed_regions.pt]: The link to the readme is wrong, it's missing the compliance part of the path.

File: security/aws/rds_publicly_accessible/README.md (hunk 9 above)
Comment (praut1) [reply security/aws/rds_publicly_accessible/README.md]: Removed mentioned part from the URL

File: compliance/aws/disallowed_regions/aws_disallowed_regions.pt (hunk 10 above)
Comment (praut1) [reply compliance/aws/disallowed_regions/aws_disallowed_regions.pt]: Updated the schemes part with "aws","aws_sts"

File: cost/aws/burstable_instance_cloudwatch_credit_utilization/aws_burstable_instance_cloudwatch_credit_utilization.pt (hunk 12 above)
Comment (praut1) [reply cost/aws/burstable_instance_cloudwatch_credit_utilization/aws_burstable_instance_cloudwatch_credit_utilization.pt]: Updated the schemes part with "aws","aws_sts"

File: cost/aws/elb/clb_unused/aws_delete_unused_clb.pt (hunk 13 above)
Comment (praut1) [reply cost/aws/elb/clb_unused/aws_delete_unused_clb.pt]: Updated the schemes part with "aws","aws_sts"

File: cost/aws/reserved_instances/coverage/reserved_instance_coverage.pt (hunk 16 above)
Comment (praut1) [reply cost/aws/reserved_instances/coverage/reserved_instance_coverage.pt]: Updated the schemes part with "aws","aws_sts"

File: operational/dbaas/aws/rds_backup/aws_rds_backup.pt (hunk 17 above)
Comment (praut1) [reply operational/dbaas/aws/rds_backup/aws_rds_backup.pt]: Updated the schemes part with "aws","aws_sts"

File: security/aws/loadbalancer_internet_facing/aws_internet-facing_elbs.pt (hunk 18 above)
Comment (praut1) [reply security/aws/loadbalancer_internet_facing/aws_internet-facing_elbs.pt]: Updated the schemes part with "aws","aws_sts"

File: security/aws/rds_publicly_accessible/aws_publicly_accessible_rds_instances.pt (hunk 19 above)
Comment (praut1) [reply security/aws/rds_publicly_accessible/aws_publicly_accessible_rds_instances.pt]: Updated the schemes part with "aws","aws_sts"

File: compliance/aws/disallowed_regions/aws_disallowed_regions.pt (hunk 20 above)
Comment (praut1) [reply compliance/aws/disallowed_regions/aws_disallowed_regions.pt]: Link to readme has been updated

File: compliance/aws/long_stopped_instances/aws_long_stopped_instances.pt (hunk 11 above)
Comment (praut1) [reply compliance/aws/long_stopped_instances/aws_long_stopped_instances.pt]: Updated the schemes part with "aws","aws_sts"

**Pull request merged**